cmd/istio-cni-daemonset/main.go

```diff
@@ -0,0 +1,363 @@
+	flag.Bool("run-as-daemon", false, "Controller will run in a loop")
+	flag.Int("daemon-poll-period", 10, "Polling period for daemon (in seconds)")
+	flag.String("broken-pod-label-key", "cni.istio.io/uninitialized", "The key portion of the label which will be set by the reconciler if --label-pods is true")
+	flag.String("broken-pod-label-value", "true", "The value portion of the label which will be set by the reconciler if --label-pods is true")
+
+	// Get kubernetes config
+	if home := homedir.HomeDir(); home != "" {
+		flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(option) absolute path to the kubeconfig file")
+	} else {
+		flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+
+	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
+	pflag.Parse()
+	if err := viper.BindPFlags(pflag.CommandLine); err != nil {
+		log.Fatal("Error parsing command line args: %+v")
+	}
+
+	// Pull runtime args into structs
+	filter = Filters{
+		InitContainerName:               viper.GetString("init-container-name"),
+		InitContainerTerminationMessage: viper.GetString("init-container-termination-message"),
+		InitContainerExitCode:           viper.GetInt("init-container-exit-code"),
+		FieldSelectors:                  &FieldSelectorSet{},
+	}
+	controlopts = ControllerOptions{
+		DeletePods:       viper.GetBool("delete-pods"),
+		RunAsDaemon:      viper.GetBool("run-as-daemon"),
+		DaemonPollPeriod: viper.GetInt("daemon-poll-period"),
+		LabelPods:        viper.GetBool("label-pods"),
+		PodLabelKey:      viper.GetString("broken-pod-label-key"),
+		PodLabelValue:    viper.GetString("broken-pod-label-value"),
+		KubeConfigPath:   viper.GetString("kubeconfig"),
+	}
+
+	filter.FieldSelectors.addFieldSelectorIfNotEmpty("spec.nodeName", viper.GetString("node_name"))
+
+	return
+}
+
+func clientSetup(options ControllerOptions) (clientset *client.Clientset, err error) {
+	var kubeConfig *rest.Config
+	// Set up client
+	if options.KubeConfigPath == "" {
+		kubeConfig, err = rest.InClusterConfig()
+		if err != nil {
+			return
+		}
+	} else {
+		kubeConfig, err = clientcmd.BuildConfigFromFlags("", options.KubeConfigPath)
+		if err != nil {
+			return
+		}
+	}
+
+	clientset, err = client.NewForConfig(kubeConfig)
+	return
+}
+
+func main() {
+	var (
+		filters = Filters{}
+		options = ControllerOptions{}
+	)
+
+	loggingOptions.OutputPaths = []string{"stderr"}
+	loggingOptions.JSONEncoding = true
+	if err := log.Configure(loggingOptions); err != nil {
+		os.Exit(1)
+	}
+
+	filters, options = parseFlags()
+	logCurrentOptions(filters, options)
+
+	// TODO:(stewartbutler) This should probably use client-go/tools/cache at some
+	//  point, but I'm not sure yet if it is needed.
+	clientSet, err := clientSetup(options)
+	if err != nil {
+		log.Fatalf("Could not construct clientSet: %s", err)
+	}
+	podFixer := brokenPodReconciler{client: *clientSet}
+
+	if options.RunAsDaemon {
+		var wg sync.WaitGroup
+		wg.Add(1)
+		go func(wg sync.WaitGroup) {
```

Comment:
It seems like this is the same as not running in a goroutine, am I missing something?

Reply:
Ah, yes -- leftover code from a refactor. At one point I had both the delete and label code running in separate goroutines, I moved them out into the single reconciliation function and forgot to remove the goroutine boilerplate. Will remove.